app/Http/Controllers/AdminController.php

```diff
@@ -178,4 +180,77 @@ public function filterListings(Request $request)
         
         return response()->json(['listings' => $listings]);
     }
+
+    /**
+     * Display a paginated list of categories in the admin panel.
+     *
+     * @return \Illuminate\View\View
+     */
+    public function categories()
+    {
+        $categories = Category::paginate(10); // Retrieve categories with pagination
```

Comment:
can you create file config/const.php and set page_pagination = 10
usage:
$page = config('const.page_pagination');